dev/com.ibm.ws.webcontainer/src/com/ibm/ws/webcontainer/srt/SRTServletRequest.java

```diff
@@ -212,6 +223,9 @@ public SRTServletRequest(SRTConnectionContext context)
         if (TraceComponent.isAnyTracingEnabled()&&logger.isLoggable (Level.FINE)) {  //306998.15
             logger.logp(Level.FINE, CLASS_NAME,"SRTServletRequest", "this->"+this+": " + "inputStream is of type --> " + this._in);
         }
+        this.threadData = SRTServletRequestThreadData.getInstance();
+        this.threadData.init(null);
+        requestThread = Thread.currentThread();
     }
```

Comment:
```suggestion
        initRequestThreadData()
    }

    protected SRTRequestContext createRequestContext() {
        return new SRTRequestContext(this);
    }

    protected void initRequestThreadData() {
        this.threadData = SRTServletRequestThreadData.getInstance();
        this.threadData.init(null);
        requestThread = Thread.currentThread();
    }
```

and also update `this._requestContext = new SRTRequestContext(this);` to `this._requestContext = createRequestContext();

Reply:
done